### PR TITLE
Add a favorites filter to movies

### PR DESCRIPTION
**Pull request by bisby**
**Changes**

Add a "Favorites" filter to the filter tab in options.

Changes it from a "Coming Soon..." message to a "All", "Favorites" in movies and just "All" in tvshows

**Issues**

Was originally going to address #276 but when I was in there, I realized it was already done, so I took a next step towards a thing my wife wanted instead.

### CODE REVIEW

**Comment by neilsb** [comment]
Looks good, although it has uncovered an issue I'd missed.   If you have no favourites it shows the "No Items" text, but when you switch back to all it doesn't hide the "No Items" text, which shows over the top of the item grid.   Nothing at all to do with the changes for this PR, just enabled me to see the issue.  😀

File: components/ItemGrid2/ItemGrid2.brs [hunk 1]
```diff
@@ -237,6 +252,13 @@ sub optionsClosed()
     m.data = CreateObject("roSGNode", "ContentNode")
     m.itemGrid.content = m.data
     loadInitialItems()
+  else if m.options.filter <> m.filter then
```
Comment (neilsb) [inline components/ItemGrid2/ItemGrid2.brs]: Having this as an if-else means that if the options were opened, and the filter was changed and the sort order was changed at the same time, then the filter change is going to be missed when the options dialog is closed I think.

Comment (bisby) [reply components/ItemGrid2/ItemGrid2.brs]: Good point. i was originally thinking of this as a "single click to pick" type feature. I updated this to check both, and then reload if a reload is necessary.

Comment (neilsb) [reply components/ItemGrid2/ItemGrid2.brs]: Great - thanks.

**Review by neilsb — APPROVED** [review]

**Pull request merged**